app/controllers/orders_controller.rb

```diff
@@ -1,26 +1,38 @@
 class OrdersController < ApplicationController
     before_action :authenticate_user!
-    before_action :set_order, only: %i[ show ]
+    before_action :set_order, only: %i[ show destroy update]
+    
+    include Pundit::Authorization
+    include OrdersHelper
 
     def create
         cart_items = current_user.cart.cart_items
         
-        order = Order.new(user: current_user)
-        total = 0
+        @order = Order.new(user: current_user)
+        total = calculate_total_order_value(cart_items)
         cart_items.each do |cart_item|
             product = Product.find(cart_item[:product_id])
-            total += cart_item.quantity * product.price
-            order.order_items.build(product: product, quantity: cart_item[:quantity],total: cart_item.quantity * product.price)
+            @order.order_items.build(product: product, quantity: cart_item[:quantity], total: cart_item.quantity * product.price)
         end
 
-        order.total = total
+        @order.total = total
 
-        if order.save
+        if @order.save
             cart_items.destroy_all
-            @items = order.order_items
-            render 'orders/show'
+            @items = @order.order_items
+            redirect_to controller: 'orders', action: 'index'
         else
-            render 'cart/index', status: :unprocessable_entity
+            flash[:error] = "Order Failed"
+            redirect_to controller: 'cart', action: 'index'
+        end
+    end
+
+    def destroy
+        authorize @order, :destroy?
+        if @order.destroy!
+            redirect_to controller:"home", action: "index"
```

Comment:
Any success message. ?

Reply:
Yaa that part is not done yet as this for admin side ,i will add a success message